pallets/author-mapping/src/lib.rs

```diff
@@ -162,11 +169,16 @@ pub mod pallet {
 			);
 
 			MappingWithDeposit::<T>::remove(&old_author_id);
-			MappingWithDeposit::<T>::insert(&new_author_id, &stored_info);
+			let new_stored_info = RegistrationInfo {
+				keys: new_author_id.clone().into(),
+				..stored_info
+			};
+			MappingWithDeposit::<T>::insert(&new_author_id, &new_stored_info);
```

Comment:
@librelois and I agree that `update_association` should set the session keys to the `new_author_id.into()` by default